File: Maximum Subarray/MaximumSubarray.py
class MaximumSubarray(object):
    def maxSubArray(self, nums):
        if len(nums) == 1:
            return nums[0]
        currentMax = nums[0]
        lastMax = nums[0]
        for i in range(1,len(nums)):
            if lastMax<0:
                lastMax = nums[i]
            else:
                lastMax += nums[i]
            currentMax = max(currentMax,lastMax)
        return currentMax

File: 0053_Maximum Subarray/MaximumSubarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if (nums == null || nums.length == 0)
            return 0;
        if (nums.length == 1)
            return nums[0];
        int ans = nums[0];
        int lastMax = nums[0];

        for (int i = 1; i < nums.length; i++) {
            if (lastMax < 0) {
                lastMax = nums[i];
            } else {
                lastMax += nums[i];
            }
            ans = Math.max(ans, lastMax);
        }

        return ans;
    }
}